### ISSUE TITLE
Obsidian Rest API starts hanging forever when patch content is called with invalid target

### ISSUE THREAD

**Issue by MarkusPfundstein** [MarkusPfundstein/mcp-obsidian#3]
When the tool patch_content is called , the ToolHandler calls the URL `PATCH /vault/{filename}`.

When the Obsidian REST API can't find the target, it throws a PatchFailed error. It seems as if this error is never returned to the client. After a while, MCP throws a timeout error and Claude tells us that it can't do it.

This is annoying but not breaking. However, the Obsidian plugin also becomes totally unresponsive afterwards. Each call then timeouts, also if done via curl (so not related to MCP).

No idea yet how to fix it. There is an issue reported here that might resolve it? https://github.com/coddingtonbear/obsidian-local-rest-api/pull/139/files

**Comment by 7shi** [comment]
I've added timeout settings in PR #4 to prevent hanging forever.

While testing, I found that:

- MCP Inspector disconnects from MCP server after ~10 seconds
- Claude Desktop App disconnects after ~8 seconds

Based on these observations, I set the timeout values to:

- Connect timeout: 3 seconds
- Read timeout: 6 seconds

This ensures that the client receives timeout errors before MCP clients force a disconnect.

**Comment by MarkusPfundstein** [comment]
love it

**Comment by 7shi** [comment]
While investigating the timeout settings (trying different values to determine the `(3,6)` parameters), I realized that this issue might be reproducible with a simple wait operation without any actual requests.

I created a minimal reproduction case that simply waits for a specified number of seconds. While looking into reporting this, I noticed that @MarkusPfundstein had already created the issue: modelcontextprotocol/python-sdk#88

The problem occurs with this simple pattern:

1. Long-running operation (> 8-10 seconds)
2. Server successfully returns results
3. Server crashes afterwards, preventing further requests

I've documented my findings in detail there.
